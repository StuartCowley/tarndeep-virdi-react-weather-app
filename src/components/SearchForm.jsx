import React from "react";

export default function SearchForm() {
  const [searchText, setSearchText] = React.useState("");

  const handleText = (event) => {
    setSearchText(event.target.value);
  };

  const handleSearch = () => {
    console.log(`Searching For ${searchText}`);
  };

  return (
    <section className="search--form">
      <input
        placeholder="Search Location..."
        type="text"
        onChange={handleText}
        value={searchText}
      />
      <button className="search--button" type="button" onClick={handleSearch}>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="30px"
          height="30px"
          viewBox="0 0 24 24"
          className="search--button-icon"
        >
          <path
            fill="white"
            d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5l-1.5 1.5l-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16A6.5 6.5 0 0 1 3 9.5A6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14S14 12 14 9.5S12 5 9.5 5Z"
          />
        </svg>
      </button>
    </section>
  );
}
