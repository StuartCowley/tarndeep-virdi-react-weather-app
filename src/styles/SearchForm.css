.search--form input[type="text"] {
  max-width: 400px;
  width: 70%;
  text-indent: 10px;
  font-family: "Lato", sans-serif;
  background-color: transparent;
  color: white;
  border: none;
  border-bottom: 2px solid white;
}

.search--form {
  padding: 1rem 0;
  display: flex;
  justify-content: center;
}

::placeholder {
  color: white;
  opacity: 1;
  font-family: "Lato", sans-serif;
}

:-ms-input-placeholder {
  color: white;
}

::-ms-input-placeholder {
  color: white;
}

.search--button {
  cursor: pointer;
  background-color: transparent;
  border: none;
}

.search--button:active {
  transform: scale(1.1);
}

.search--button:hover {
  background-color: #6ea4bfff;
  border-radius: 4px;
}

.search--button-icon {
  display: block;
  background-color: transparent;
}
