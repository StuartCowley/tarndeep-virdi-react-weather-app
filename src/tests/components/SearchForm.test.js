import React from "react";
import { render, fireEvent } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  test("renders the input field and button", () => {
    const { getByPlaceholderText, getByRole } = render(<SearchForm />);
    const inputElement = getByPlaceholderText("Search Location...");
    const buttonElement = getByRole("button");
    expect(inputElement).toBeInTheDocument();
    expect(buttonElement).toBeInTheDocument();
    expect({ getByPlaceholderText, getByRole }).toMatchSnapshot();
  });

  test("calls onSearch prop when the button is clicked", () => {
    const mockOnSearch = jest.fn();
    const { getByRole, getByDisplayValue } = render(
      <SearchForm onSearch={mockOnSearch} />
    );
    const inputElement = getByDisplayValue("");
    const buttonElement = getByRole("button");
    fireEvent.change(inputElement, { target: { value: "London" } });
    fireEvent.click(buttonElement);
    expect(mockOnSearch).toHaveBeenCalledWith("London");
  });

  test("updates the search text when input value changes", () => {
    const { getByDisplayValue } = render(<SearchForm />);
    const inputElement = getByDisplayValue("");
    fireEvent.change(inputElement, { target: { value: "New York" } });
    expect(inputElement.value).toBe("New York");
  });
});
