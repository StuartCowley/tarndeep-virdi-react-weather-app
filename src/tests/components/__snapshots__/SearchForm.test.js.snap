// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders the input field and button 1`] = `
Object {
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
}
`;
